From ec032370f2459544a0b5e2b995699dab29e580ef Mon Sep 17 00:00:00 2001
From: Kevin Burke <kevin@meter.com>
Date: Sun, 12 Jul 2020 14:35:27 -0700
Subject: [PATCH] helper/schema: ensure test works with Go 1.15 error messages

I verified that the code itself does not have similar assertions about
the error message that would break if a different string was returned
for the error message.

Fixes #25553.
---
 helper/schema/provider_test.go | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/helper/schema/provider_test.go b/helper/schema/provider_test.go
index 01ee82a1f4d..13809b35bbf 100644
--- a/helper/schema/provider_test.go
+++ b/helper/schema/provider_test.go
@@ -3,7 +3,7 @@ package schema
 import (
 	"fmt"
 	"reflect"
-	"strings"
+	"regexp"
 	"testing"
 	"time"
 
@@ -341,9 +341,11 @@ func TestProviderDiff_timeoutInvalidValue(t *testing.T) {
 	if err == nil {
 		t.Fatal("Expected provider.Diff to fail with invalid timeout value")
 	}
-	expectedErrMsg := "time: invalid duration invalid"
-	if !strings.Contains(err.Error(), expectedErrMsg) {
-		t.Fatalf("Unexpected error message: %q\nExpected message to contain %q",
+	// Go 1.15 added quotes around the invalid duration value. You can remove
+	// the regex once Go 1.14 and below are no longer supported.
+	expectedErrMsg := regexp.MustCompile(`time: invalid duration "?invalid"?`)
+	if !expectedErrMsg.MatchString(err.Error()) {
+		t.Fatalf("Unexpected error message: %q\nExpected message to match rx %q",
 			err.Error(),
 			expectedErrMsg)
 	}
